xbmc/GUIPassword.cpp

```diff
@@ -377,29 +377,29 @@ bool CGUIPassword::CheckSettingLevelLock(const SettingLevel& level, bool enforce
 {
   const std::shared_ptr<CProfileManager> profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
 
-  LOCK_LEVEL::SETTINGS_LOCK lockLevel = profileManager->GetCurrentProfile().settingsLockLevel();
+  SettingsLock lockLevel = profileManager->GetCurrentProfile().settingsLockLevel();
 
-  if (lockLevel == LOCK_LEVEL::NONE)
+  if (lockLevel == SettingsLock::NONE)
     return true;
 
     //check if we are already in settings and in an level that needs unlocking
   int windowID = CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow();
-  if ((int)lockLevel-1 <= (short)CViewStateSettings::GetInstance().GetSettingLevel() &&
+  if (static_cast<short>(lockLevel) - 1 <= static_cast<short>(CViewStateSettings::GetInstance().GetSettingLevel()) &&
```

Comment:
why not use int? enums are int by default
```suggestion
  if (static_cast<int>(lockLevel) - 1 <= static_cast<int>(CViewStateSettings::GetInstance().GetSettingLevel()) &&
```

also others on below code

Reply:
shorts changed to int.